pygbm/predictor.py

```diff
@@ -13,6 +13,11 @@
     ('right', np.uint32),
     ('gain', np.float32),
     ('depth', np.uint32),
+    ('use_sub', np.bool),
+    ('time', np.float32),
+    ('ratio', np.float32),
+    ('sum_g', np.float32),
+    ('sum_h', np.float32),
```

Comment:
Do we really need  to store the sum of gradients and hessians in the predictor? Maybe we should find a way to just plot the grower tree instead of the predictor tree if the goal is to present a tree-structured profile report of the cost of growing a tree.

The predictor tree is part of the public API of the model and I would rather note pollute it with attributes that are only useful for pygbm developers who try to optimize the implementation.

Reply:
No we don't need those. Same goes for `time` and `ratio`, etc. Those were only useful for debugging.

I agree that we should instead plot the grower tree. That's quite easy to do I think, we would just need to keep a `growers_` list attribute in the `GradientBoostingMachine` class and slightly change the plotting procedure. Should I do this? Here or another PR?

Reply:
Plotting the predictor trees is still useful for the users to learn about the structure of the trees. So the plotting utility should still work in that case.

I just want to extend it to make it work with additional info on the grower tree and use it in an example folder stating explicitly that this is only to analyze the performance profile of a single small tree. E.g.:

```
example/plot_performance_profile_single_small_tree.py
```